modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/admin/application_authorizations.jsp

```diff
@@ -22,6 +22,11 @@ String redirect = ParamUtil.getString(request, "redirect");
 portletDisplay.setShowBackIcon(true);
 portletDisplay.setURLBack(redirect);
 
+// Instead of doing a return here we should have an MVCRenderCommand that checks
+// this permissions and decides whether or not we should dispatch to the
+// application_authorization.jsp or to some error.jsp, but we shouldn't have a
+// return here.
+
 if (!oAuth2AdminPortletDisplayContext.hasViewGrantedAuthorizationsPermission()) {
```

Comment:
Thanks @sergiogonzalez, how can we check permission inside JSP when it's accessed directly using mvcPath request parameter, i.e. skip MVCRenderCommand? Make sure there's NPE in JSP and it doesn't render?

Thanks!

Reply:
OK, so NPE covers that :)